charts/kong/values.yaml

```
@@ -581,6 +581,68 @@ postgresql:
+# cert-manager certificate issuer parameters
+# -----------------------------------------------------------------------------
+
+# Cert Manager Documentation:
+# https://cert-manager.io/docs/
+
+# Kong services and proxied services can have various certificate requirements.
+# By default, this chart installs Kong without cert-manager and generates
+# a fallback self-signed certificate on each proxy dataplane.
+# This cert-manager support requires that cert-manager be installed and ready
+# before deploying Kong.
+
+# These are the options for using the cert-manager integration:
+#
+# - (recommended) Deploy cert-manager and configure valid ssl issuer.
+#   Leverage your valid ssl issuer for generating any or all of the following:
+#   - kong proxy default certificate
+#   - database service certificate
+#   - kong services certificates
+#
+# - Alternatively use the built in self-signed issuer to issue self-signed
+#   certificates for kong hybrid cluster mtls. Use other valid ssl issuer for all
+#   other certificate needs.
+#
+# - Finally, use the built in self-signed issuer to provision all certificates.
+#   This method is generally only recommended for the following scenarios:
+#   - Kong utilizes self-signed certs behind an SSL terminating LoadBalancer
+#     which handles public traffic with valid ssl certificates and only relies
+#     on the self-signed certs on upstream connections between the loadbalancer
+#     and the kong proxy and services directly.
+#   - Kong helm chart cert-manager integration is used only for provisioning
+#     the cluster mtls shared certificate for use in kong hybrid and multiple-dataplane
+#     scenarios.
+#   - Kong pre-prod & local development where the self-signed cert-manager issuer
+#     is leveraged for developer and testing workflows only. In this scenario it
+#     is assumed that valid certs will be utilized in production for public facing
+#     traffic.
+
+# NOTICE: The cert-manager cluster cert integration generates a self signed
+# cluster issuer certificate authority for hybrid cluster mtls when enabled.
+
+certificate:
+  # Set cluster.enabled to true to issue kong hybrid mtls certificate with cert-manager
+  cluster:
+    enabled: false
+  # Set proxy.enabled to true to issue default kong-proxy certificate with cert-manager
+  proxy:
+    enabled: false
+    # Set `issuer` to name of alternate cert-manager clusterIssuer to override default
+    # self-signed issuer. Note: certificate.cluster.enabled must be set to true if this
+    # is not defined with a valid alternative.
+    issuer: ""
+    # Use commonName and dnsNames to set the common name and dns alt names which this
+    # certificate is valid for. Wildcard records are supported by the included self-signed issuer.
+    commonName: "example.com"
+    dnsNames:
+    - "example.com"
+    - "*.example.com"
+    - "*.apps.example.com"
+    - "*.kong.example.com"
```

Default array values will actually get used, so those should be comment examples with `{}` as the actual default. Using `example.com` versus the `.example` TLD is basically personal preference, but I like the latter a bit more if we're going to use actual strings rather than commenting those out as well (not sure if leaving it empty would break the resulting Certificate resources, but ambivalent since you have to enable this anyway and presumably will notice that you also need to change the CN).

```suggestion
    commonName: "kong.example"
    # Remove the "{}" and uncomment/change the examples to add SANs
    dnsNames:{}
    #- "kong.example"
    #- "*.kong.example"
```